src/background/host.ts

```diff
@@ -34,23 +36,148 @@ export class Host {
         });
 
         port.onMessage.addListener(message => {
-            this.#friends.forEach(conn => conn.send(message));
+            this.#friends.forEach(friend => friend.sendMessage(message));
         });
 
         this.#peer.on("connection", (conn: Peer.DataConnection) => {
-            this.#handleConnect(conn);
-            // for incoming connections, poll our video and transmit the status
-            setTimeout(() => port.postMessage({ type: MessageType.Poll }), 500);
+            this.onPeerConnectToHost(conn);
         });
     }
 
-    public connect(hostId: string): void {
+    /**
+     * Subscribe to friend changes
+     * @param cb The callback to call when the friends list changes
+     */
+    public subscribeToFriendChanges(cb: (count: number, delta: number) => void): void {
+        this.friendChangeSubscribers.push(cb);
+    }
+
+    /**
+     * Connects the current host to the given peer
+     * @param hostId The id of the host to connect to
+     */
+    public connectToHost(hostId: string): void {
         const conn = this.#peer.connect(hostId);
 
-        conn.on("open", () => this.#handleConnect(conn));
+        // Handle the conenction to host but don't notify friend changes until the host sends the friend message
+        conn.on("open", () => this.handleConnect(conn, false));
```

Comment:
I'm really confused by this class now. It's named `Host`, and we have a `Friend` class now ... but every peer (i.e. Friend) will have an instance of `new Host()` stored. So ... It's really hard to keep straight who the host is and who the friend is.

Reply:
Yeah I agree we should change the name but didn't want to make the code review bigger. I was thinking there's basically one host instance per person (computer) but each person (computer) will have N friend instances

Reply:
I think instead of Host it should be named like StreamPeerManager or something similar